### PR TITLE
algorithm_comparison maintenance 

### PR DESCRIPTION
**Pull request by burner**
* selective import
* some symbols where used only private
* missing ditto

### CODE REVIEW

File: std/algorithm/comparison.d [hunk 1]
```diff
@@ -560,7 +560,7 @@ if (isInputRange!R1 && isInputRange!R2 && !(isSomeString!R1 && isSomeString!R2))
     }
 }
 
-// Specialization for strings (for speed purposes)
+/// Ditto Specialization for strings (for speed purposes)
 int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2) if (isSomeString!R1 && isSomeString!R2)
```
Comment (JakobOvrum) [inline std/algorithm/comparison.d]: Does making this show up in documentation really add any documentation value? Also, `/// Ditto` *and* a unique description in the same comment? How does that work?

Comment (burner) [inline std/algorithm/comparison.d]: my bad

File: std/algorithm/comparison.d [hunk 2]
```diff
@@ -560,7 +560,7 @@ if (isInputRange!R1 && isInputRange!R2 && !(isSomeString!R1 && isSomeString!R2))
     }
 }
 
-// Specialization for strings (for speed purposes)
+/// Specialization for strings (for speed purposes)
```
Comment (JakobOvrum) [inline std/algorithm/comparison.d]: Like this it will turn up in its own section... with a really sparse description. That would look pretty bad...

Comment (burner) [inline std/algorithm/comparison.d]: I looked at the output. It really gets the point across.

Comment (JakobOvrum) [inline std/algorithm/comparison.d]: And Ddox?

Comment (burner) [inline std/algorithm/comparison.d]: I don not get what you ask? please eleborate

Comment (quickfur) [inline std/algorithm/comparison.d]: Shouldn't this just be `/// ditto`? The docs for the previous overload of `cmp` already describes the behaviour with string arguments.

**Comment by quickfur** [comment]
Also, the unittest contains an extraneous `printf("string.cmp.unittest\n");` that really should not be displayed in the documentation. It looks like remnants of some debugging session a long time ago. In any case, it has no place in user documentation; please delete that line.

File: std/algorithm/comparison.d [hunk 3]
```diff
@@ -523,7 +523,7 @@ body
     static assert(is(typeof(clamp(-1L, -2L, 2UL)) == long));
 
     // user-defined types
-    import std.datetime;
+    import std.datetime : Date;
```
Comment (andralex) [inline std/algorithm/comparison.d]: meh - not really improving things

Comment (burner) [inline std/algorithm/comparison.d]: please explain

**Comment by burner** [comment]
thanks

**Pull request merged**